src/main/java/org/truffleruby/core/string/TBytesKey.java

```diff
@@ -62,4 +91,51 @@ public String toString() {
         return TruffleString.fromByteArrayUncached(bytes, encoding, false).toString();
     }
 
+    private static int hashCode(InternalByteArray byteArray) {
```

Comment:
I think it would be nice if we could upstream this into Truffle's `ArrayUtils`.

Reply:
That'd be fine, too. There had been some concerns previously about exposing public APIs in something named "internal", but I don't really care where it goes.